astronomer/providers/snowflake/hooks/snowflake.py

```diff
@@ -88,7 +88,7 @@ def run(  # type: ignore[override]
                         cur.execute_async(sql_statement)
                     query_id = cur.sfqid
                     self.log.info("Snowflake query id: %s", query_id)
-                    self.query_ids.append(query_id)
+                    self.query_ids.append(query_id)  # type: ignore[arg-type]
```

Comment:
I think we should avoid ignoring the type incompatibility error. In general, those errors help us to identify potential issues in our code, no?

I am looking at the parent hook and it defines self.query_ids type to be list[str]. But query_id = cur.sfqid could be None too. Hence, I believe we should not append None query_id values to self.query_ids, because we would not be able to check the statuses of such queries anyway with the None ID. 

```suggestion
                    if query_id:
                        self.query_ids.append(query_id)
```